packages/amplify-provider-awscloudformation/src/utils/auth-types.ts

```diff
@@ -1,3 +1,23 @@
+import { $TSAny } from 'amplify-cli-core';
+
+export type AuthFlow = 'admin' | 'profile' | 'accessKeys';
+export interface AuthFlowConfig extends Partial<AwsSdkConfig> {
+  type: AuthFlow;
+  appId?: string;
+  profileName?: string;
+  useProfile?: boolean;
+}
+
+export interface AwsSdkConfig {
+  accessKeyId: string;
+  expiration?: Date;
+  region: string;
+  secretAccessKey: string;
+  sessionToken?: string;
+  httpOptions?: {
+    agent: $TSAny;
```

Comment:
could a more restrictive type be used here?

Reply:
Not trivially. I felt it was out of scope for the PR because of all the downstream effects.